File: desktop/cmd/personal/aiembed_other.go
//go:build !windows && !darwin

package main

import "github.com/wailsapp/wails/v3/pkg/application"

// aiembed_other.go is the fallback for platforms without native window embedding
// (currently Linux). Windows uses Win32 owner windows (aiembed_windows.go) and
// macOS uses NSWindow child windows (aiembed_darwin.go); on the rest we don't
// reparent and let ActivateAIService fall back to opening each service in its own
// independent window (see aiwin.go's aiEmbedSupported branch). These stubs keep
// the shared code compiling on every platform.

const aiEmbedSupported = false

func aiEmbedOwn(hub, child *application.WebviewWindow) {}

func aiEmbedPosition(hub, child *application.WebviewWindow, x, y, w, h int) {}

func aiEmbedReveal(hub, child *application.WebviewWindow, x, y, w, h int) {}

func aiEmbedStash(child *application.WebviewWindow, w, h int) {}

File: desktop-v2/src/ai_embed_other.rs
//! AI 服务窗口嵌入（非 Windows stub）：macOS/Linux 上 v1 也是独立普通窗口
//! （aiEmbedSupported=false，见 aiembed_other.go），v2 对齐——服务窗口用普通
//! show/hide 语义，不做 Win32 停靠。窗口管理在 commands.rs 里按本常量分流。

use tauri::WebviewWindow;

pub const AI_EMBED_SUPPORTED: bool = false;

// 这些函数在 Windows 版里做 Win32 停靠（ai_embed_windows.rs）；非 Windows 是
// 普通窗口语义，commands.rs 在 AI_EMBED_SUPPORTED 分支下不调用它们，但引用
// 处是运行时判断而非 #[cfg]，所以这里必须提供同名 no-op stub 保持可编译。
pub fn dock_over_stage(_hub: &WebviewWindow, _child: &WebviewWindow) {}
pub fn reveal_over_stage(_hub: &WebviewWindow, _child: &WebviewWindow, _stage: (i32, i32, i32, i32)) {}
pub fn position_window(_hub: &WebviewWindow, _child: &WebviewWindow, _stage: (i32, i32, i32, i32)) {}
pub fn stash_offscreen(_win: &WebviewWindow, _stage: (i32, i32, i32, i32)) {}
